hera_cal/vis_clean.py

```diff
@@ -7,29 +7,240 @@
+            ind_right = np.max(unflagged_chans) + 1
+        dout = data_in[:, ind_left: ind_right]
+        wout = weights_in[:, ind_left: ind_right]
+        edges = (ind_left, data_in.shape[1] - ind_right)
+        if ax == 'both':
+            x1 = x[1][ind_left: ind_right]
+            x0, dout, wout, e0 = truncate_flagged_edges(dout, wout, x[0], ax='time')
+            xout = [x0, x1]
+            edges = [e0[0], edges]
+        else:
+            xout = x[ind_left: ind_right]
+            edges = [(0, 0), edges]
+    return xout, dout, wout, edges
+
+
+def flag_rows_with_flags_within_edge_distance(weights_in, min_flag_edge_distance, ax='freq'):
+    """
+    flag integrations (and/or channels) with flags within min_flag_edge_distance of edge.
+
+    Parameters
+    -----------
+    weights_in : array-like, 2d
+        weights to check for flags within min_edge distance of edge along specified axis.
+        will set all weights in each row with flags within min_flag_edge_distance to zero.
+    min_flag_edge_distance : integer (or two-tuple / list)
+        any row of weights_in with zero weights within min_edge distance
+        of edge will be set to zero.
+    ax : str, optional
+        string specifying which axis to flag edges of.
+        valid options include 'freq', 'time', 'both'.
+        default is 'freq'
+    Returns
+    -------
+    wout, array-like 2d
+        weights with rows or columns with zero weights
+        within min_flag_edge_distance set entirely to zero.
+
+    """
+    if ax == 'time':
+        wout = flag_rows_with_flags_within_edge_distance(weights_in.T, min_flag_edge_distance).T
+    else:
+        wout = copy.deepcopy(weights_in)
+        for rownum, wrow in enumerate(wout):
+            if ax == 'both':
+                if np.any(np.isclose(wout[rownum, :min_flag_edge_distance[1]], 0.0)) | np.any(np.isclose(wout[rownum, -min_flag_edge_distance[1] - 1:], 0.0)):
+                    wout[rownum, :] = 0.
+            elif np.any(np.isclose(wout[rownum, :min_flag_edge_distance], 0.0)) | np.any(np.isclose(wout[rownum, -min_flag_edge_distance - 1:], 0.0)):
+                wout[rownum, :] = 0.
+        if ax == 'both':
+            wout = flag_rows_with_flags_within_edge_distance(wout, min_flag_edge_distance[0], ax='time')
+    return wout
+
+
+def flag_rows_with_contiguous_flags(weights_in, max_contiguous_flag, ax='freq'):
+    """
+    flag any row or column with contiguous zero-weights over limit.
+
+    Parameters
+    ----------
+    weights_in : array-like, 2d
+        weights to check. any row (ax='time') or col (ax='freq')
+        with contiguous regions of zero with length greater then max_contiguous_flag
+        will be set to zero.
+    max_contiguous_flag : integer (or 2-list/tuple)
+        any row or column with contiguous weights equal to zero greater or equal to this value
+        will be set entirely to zero.
+    ax : str, optional
+        axis to perform flagging over.
```

Comment:
`axis to perform flagging over. options=['time', 'freq', 'both'], default='freq'`